app/controllers/concerns/hydranorth/files_controller_behavior.rb

```diff
@@ -58,6 +58,16 @@ def edit_form
       end
     end
 
+    def update_metadata
+      if @generic_file[:resource_type].include? Sufia.config.special_types['cstr']
+        self.edit_form_class = Hydranorth::Forms::CstrEditForm
+      elsif @generic_file[:resource_type].include? Sufia.config.special_types['ser']
+        self.edit_form_class = Hydranorth::Forms::SerEditForm
+      elsif @generic_file[:resource_type].include? Sufia.config.special_types['thesis']
+        self.edit_form_class = Hydranorth::Forms::ThesisEditForm
+      end
+      super
+    end
```

Comment:
@mbarnett good suggestions.  I'm trying this now.  Might be some bleed over into the batch controller.